code/datums/actions/species_actions/sectoid_action.dm

```diff
@@ -0,0 +1,465 @@
+		if(!silent)
+			owner.balloon_alert(owner, "Out of sight!")
+		return FALSE
+	if(carbon_target.stat == DEAD)
+		if(!silent)
+			carbon_target.balloon_alert(owner, "already dead")
+		return FALSE
+
+/datum/action/ability/activable/sectoid/mindmeld/use_ability(atom/target)
+	var/mob/living/carbon/carbon_owner = owner
+	melded_mob = target
+	melded_mob.balloon_alert_to_viewers("mindmelded")
+	owner.balloon_alert_to_viewers("mindmelded")
+	playsound(melded_mob, 'sound/effects/off_guard_ability.ogg', 50)
+
+	melded_mob.apply_status_effect(STATUS_EFFECT_MINDMEND, carbon_owner, max_range, accuracy_mod, health_mod, speed_mod, stun_resistance)
+	carbon_owner.apply_status_effect(STATUS_EFFECT_MINDMEND, carbon_owner, max_range, accuracy_mod, health_mod, speed_mod, stun_resistance)
+	RegisterSignal(melded_mob, COMSIG_MOB_DEATH, PROC_REF(end_ability))
+	RegisterSignal(carbon_owner, COMSIG_MOB_DEATH, PROC_REF(end_ability))
+
+	succeed_activate()
+	add_cooldown()
+
+/// Ends the ability if the Enhancement buff is removed.
+/datum/action/ability/activable/sectoid/mindmeld/proc/end_ability()
+	SIGNAL_HANDLER
+	var/mob/living/carbon/carbon_owner = owner
+	add_cooldown()
+	carbon_owner.remove_status_effect(STATUS_EFFECT_MINDMEND)
+	if(!melded_mob)
+		return
+	melded_mob.remove_status_effect(STATUS_EFFECT_MINDMEND)
+	melded_mob = null
+
+/datum/action/ability/activable/sectoid/mindmeld/greater
+	name = "Greater Mindmeld"
+	desc = "Merge minds with the target, greatly empowering both."
+	max_range = 12
+	accuracy_mod = 40
+	health_mod = 70
+	speed_mod = -0.5
+	stun_resistance = 50
+
+#define MINDFRAY_RANGE 8
+/datum/action/ability/activable/sectoid/mindfray
+	name = "Mindfray"
+	action_icon_state = "mindfray"
+	desc = "Muddles the mind of an enemy, making it harder for them to focus their aim for a while."
+	cooldown_duration = 20 SECONDS
+	target_flags = ABILITY_MOB_TARGET
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_ABILITY_MINDFRAY,
+	)
+	///damage of this ability
+	var/damage = 20
+
+/datum/action/ability/activable/sectoid/mindfray/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return
+	if(!iscarbon(A))
+		if(!silent)
+			A.balloon_alert(owner, "not living")
+		return FALSE
+	if(!line_of_sight(owner, A, 9))
+		if(!silent)
+			owner.balloon_alert(owner, "Out of sight!")
+		return FALSE
+	if((A.z != owner.z) || get_dist(owner, A) > MINDFRAY_RANGE)
+		if(!silent)
+			A.balloon_alert(owner, "too far")
+		return FALSE
+	var/mob/living/carbon/carbon_target = A
+	if(carbon_target.stat == DEAD)
+		if(!silent)
+			carbon_target.balloon_alert(owner, "already dead")
+		return FALSE
+
+/datum/action/ability/activable/sectoid/mindfray/use_ability(atom/target)
+	var/mob/living/carbon/carbon_target = target
+	carbon_target.apply_status_effect(STATUS_EFFECT_GUN_SKILL_SCATTER_DEBUFF, 10 SECONDS)
+	carbon_target.apply_status_effect(STATUS_EFFECT_CONFUSED, 40)
+	carbon_target.apply_damage(damage, BURN, updating_health = TRUE)
```

Comment:
it doesn't ignore armour. The damage is only really supposed to be minor, considering there is effectively nothing you can really do to avoid mindfray. Its mainly for the debuffs.